arch/cortex-m/src/dcb.rs

```diff
@@ -0,0 +1,58 @@
+//! ARM Debug Control Block
+//!
+//! <https://developer.arm.com/documentation/ddi0403/latest>
+
+use kernel::utilities::registers::interfaces::ReadWriteable;
+use kernel::utilities::registers::{register_bitfields, register_structs, ReadWrite};
+use kernel::utilities::StaticRef;
+
+//TODO: Rest of the Registers
+register_structs! {
+    DcbRegisters{
+        (0x00 => demcr: ReadWrite<u32, DebugExceptionAndMonitorControl::Register>),
+
+        (0x04 => @END),
+    }
+}
+
+register_bitfields![u32,
+    DebugExceptionAndMonitorControl [
+        /// Write 1 to globally enable all DWT and ITM features.
+        /// RW.
```

Comment:
This whole register is declared `ReadWrite` already; are the `RO/WO` fields? If not, these `RW`s should probably be removed.